rules/CodeQuality/Naming/MethodCallToVariableNameResolver.php

```diff
@@ -100,6 +100,10 @@ private function getVariableName(MethodCall $methodCall, string $methodCallVarNa
 
     private function getFallbackVarName(string $methodCallVarName, string $methodCallName): string
     {
+        $lastBackslashPos = strrpos($methodCallVarName, '\\');
+        if ($lastBackslashPos !== false) {
```

Comment:
What about `str_starts/ends_with()` with?

Reply:
The `\\` is in the middle of word, so I think it can use use `str_contains` ref https://3v4l.org/CjdNU

Reply:
In next process inside if, it tried to get last backslash + 1 to make last substring on last `\\`, so I think the `strrpos` actually ok

Reply:
I suggested the changed for better readability. It's unclear to me what the code does now.

Reply:
The another solution can use explode and then array_pop, eg:

```php
$paths  = explode('\\', $methodCallVarName);
$methodCallVarName = array_pop($paths);
```

that, it always catched ref https://3v4l.org/6QAok . What do you think ?

Reply:
function call can actually has `\\` without a `use function`, but `use`  only, eg:

```php
use A\B;

A\B\c('test');
```

which `c` is a function.

Reply:
So the `$methodCallVarName` is actually a `$functionCallName`? I'm confused

Reply:
If the goal is to get the last part from a namespaced function name, this would be more clear code:

```php
if ($this->isNamespacedFunctionName($functionCallName)) {
    return Strings::after('A\B\c', '\\', -1);
}



// ...

private function isNamespacedFunctionName(string $functionname): bool
{
    return str_contains($functionCallName, '\\');
}
```

Reply:
that's a variable name that is created with next appended. I created new PR https://github.com/rectorphp/rector-src/pull/251 for alternative with cherry-pick @freiondrej-economia commit.

Reply:
@TomasVotruba ok, updated at https://github.com/rectorphp/rector-src/pull/251 https://github.com/rectorphp/rector-src/pull/251/commits/9853f3ef8d2ff6337d859e2f9360bd4628fb251a